libraries/rush-lib/src/logic/pnpm/PnpmLockfileConfiguration.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+import { AlreadyReportedError } from '@rushstack/node-core-library';
+import fs from 'fs';
+import yaml from 'js-yaml';
+import type { Lockfile } from '@pnpm/lockfile-types';
+import type { ITerminal } from '@rushstack/terminal';
+
+import type { RushConfiguration } from '../../api/RushConfiguration';
+import { CustomTipId } from '../../api/CustomTipsConfiguration';
+import type { PnpmLockfilePolicy } from './PnpmOptionsConfiguration';
+
+export class PnpmLockfileConfiguration {
+  /**
+   * Determine whether `pnpm-lock.yaml` complies with the rules specified in `common/config/rush/pnpm-config.schema.json`.
+   * @internal
+   */
+  public static pnpmLockValidation(
```

Comment:
If this is touching the filesystem, make it async.

Reply:
I think FileSystem.readFile is a synchronous function?

Reply:
`FileSystem.readFileAsync()` is the `async` version.  (Each API has both sync and async versions.)

Reply:
> `FileSystem.readFileAsync()` is the `async` version. (Each API has both sync and async versions.)

I see, but I use sync version, right?